syntax = "proto3";

package bgpinfo;

service bgp_info {
    rpc add_latest(values) returns (result);
}

message values {
    // UTC in UNIX time
    int64 time = 1;
    prefix_count prefix_count = 2;
    repeated peer_count peers = 3;
    // Any single line used for tweet info is marked
    bool tweet = 4;
    repeated memory mem_use = 5;
    as_count as_count = 6;
    repeated mask masks = 7;
}

message result {
    bool success = 1;
}

// as_count will hold the total amount of unique ASs
message as_count {
    int32 as4_len = 1;
    int32 as6_len = 2;
}

// total memory used by the daemon
message memory {
    address_family family = 1;
    string tables = 2;
    string total = 3;
    string protocols = 4;
    string attributes = 5;
    string roa = 6;
}

// how many peers do I have
message peer_count {
    address_family family = 1;
    int32 configured = 2;
    int32 up = 3;
}

enum address_family {
    UNKNOWN = 0;
    IPV6 = 1;
    IPV4 = 2;
}

// prefix counts. Most used info
message prefix_count {
    address_family family = 1;
    int32 total = 2;
    int32 active = 3;
}

// how many subnets of each mask is active
message mask {
    address_family address_family = 1;
    int32 mask = 2;
    int32 active = 3;
}
